Tests fail on Python 3.14

```
_________________ ERROR collecting tests/test_async_remote.py __________________
tests/test_async_remote.py:28: in <module>
    ED_APPS_LAUNCH_FUTURE = create_future_with_result(ED_APPS_LAUNCH_SAMPLE)
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
tests/test_async_remote.py:23: in create_future_with_result
    future = asyncio.Future()
             ^^^^^^^^^^^^^^^^
/nix/store/pmb8clh66mryha2ijzz3dg969drrkgj7-python3-3.14.2/lib/python3.14/asyncio/events.py:715: in get_event_loop
    raise RuntimeError('There is no current event loop in thread %r.'
E   RuntimeError: There is no current event loop in thread 'MainThread'.
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -33,6 +33,7 @@ jobs:
       fail-fast: false
       matrix:
         include:
+          - { python-version: "3.14", os: ubuntu-latest, session: "tests" }
           - { python-version: "3.13", os: ubuntu-latest, session: "tests" }
           - { python-version: "3.12", os: ubuntu-latest, session: "tests" }
           - { python-version: "3.11", os: ubuntu-latest, session: "tests" }

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -1,6 +1,5 @@
 """Tests for remote module."""
 
-import asyncio
 from unittest.mock import Mock, patch
 
 from aioresponses import aioresponses
@@ -31,9 +30,7 @@ def get_connection():
 @pytest.fixture(autouse=True)
 def override_asyncio_sleep():
     """Ignore asyncio sleep in tests."""
-    sleep_future = asyncio.Future()
-    sleep_future.set_result(None)
-    with patch("samsungtvws.async_connection.asyncio.sleep", return_value=sleep_future):
+    with patch("samsungtvws.async_connection.asyncio.sleep"):
         yield
 
 
@@ -41,10 +38,11 @@ def override_asyncio_sleep():
 def get_async_connection():
     """Open a websockets connection."""
     connection = Mock(ClientConnection)
-    with patch(
-        "samsungtvws.async_connection.connect", return_value=asyncio.Future()
-    ) as connection_class:
-        connection_class.return_value.set_result(connection)
+
+    async def _connect(*args, **kwargs):
+        return connection
+
+    with patch("samsungtvws.async_connection.connect", _connect):
         yield connection
 
 

--- a/tests/test_async_remote.py
+++ b/tests/test_async_remote.py
@@ -25,20 +25,13 @@ def create_future_with_result(result) -> asyncio.Future:
     return future
 
 
-ED_APPS_LAUNCH_FUTURE = create_future_with_result(ED_APPS_LAUNCH_SAMPLE)
-ED_EDENTV_UPDATE_FUTURE = create_future_with_result(ED_EDENTV_UPDATE_SAMPLE)
-ED_INSTALLED_APP_FUTURE = create_future_with_result(ED_INSTALLED_APP_SAMPLE)
-MS_CHANNEL_CONNECT_FUTURE = create_future_with_result(MS_CHANNEL_CONNECT_SAMPLE)
-MS_ERROR_EVENT_FUTURE = create_future_with_result(MS_ERROR_SAMPLE)
-MS_VOICEAPP_HIDE_FUTURE = create_future_with_result(MS_VOICEAPP_HIDE_SAMPLE)
-NONE_FUTURE = create_future_with_result(None)
-
-
 @pytest.mark.asyncio
 async def test_connect(async_connection: Mock) -> None:
     """Ensure simple data can be parsed."""
-    async_connection.recv = Mock(side_effect=[MS_CHANNEL_CONNECT_FUTURE])
-    async_connection.send = Mock(return_value=NONE_FUTURE)
+    async_connection.recv = Mock(
+        side_effect=[create_future_with_result(MS_CHANNEL_CONNECT_SAMPLE)]
+    )
+    async_connection.send = Mock(return_value=create_future_with_result(None))
     tv = SamsungTVWSAsyncRemote("127.0.0.1")
     await tv.start_listening()
     assert tv.token == 123456789
@@ -49,12 +42,12 @@ async def test_connect_with_extra_event(async_connection: Mock) -> None:
     """Ensure simple data can be parsed."""
     async_connection.recv = Mock(
         side_effect=[
-            MS_VOICEAPP_HIDE_FUTURE,
-            ED_EDENTV_UPDATE_FUTURE,
-            MS_CHANNEL_CONNECT_FUTURE,
+            create_future_with_result(MS_VOICEAPP_HIDE_SAMPLE),
+            create_future_with_result(ED_EDENTV_UPDATE_SAMPLE),
+            create_future_with_result(MS_CHANNEL_CONNECT_SAMPLE),
         ]
     )
-    async_connection.send = Mock(return_value=NONE_FUTURE)
+    async_connection.send = Mock(return_value=create_future_with_result(None))
     tv = SamsungTVWSAsyncRemote("127.0.0.1")
     await tv.start_listening()
     assert tv.token == 123456789
@@ -63,8 +56,10 @@ async def test_connect_with_extra_event(async_connection: Mock) -> None:
 @pytest.mark.asyncio
 async def test_connection_failure(async_connection: Mock) -> None:
     """Ensure simple data can be parsed."""
-    async_connection.recv = Mock(side_effect=[MS_ERROR_EVENT_FUTURE])
-    async_connection.send = Mock(return_value=NONE_FUTURE)
+    async_connection.recv = Mock(
+        side_effect=[create_future_with_result(MS_ERROR_SAMPLE)]
+    )
+    async_connection.send = Mock(return_value=create_future_with_result(None))
     tv = SamsungTVWSAsyncRemote("127.0.0.1")
     with pytest.raises(ConnectionFailure):
         await tv.start_listening()
@@ -73,8 +68,10 @@ async def test_connection_failure(async_connection: Mock) -> None:
 @pytest.mark.asyncio
 async def test_send_key(async_connection: Mock) -> None:
     """Ensure simple data can be parsed."""
-    async_connection.recv = Mock(side_effect=[MS_CHANNEL_CONNECT_FUTURE])
-    async_connection.send = Mock(return_value=NONE_FUTURE)
+    async_connection.recv = Mock(
+        side_effect=[create_future_with_result(MS_CHANNEL_CONNECT_SAMPLE)]
+    )
+    async_connection.send = Mock(return_value=create_future_with_result(None))
     tv = SamsungTVWSAsyncRemote("127.0.0.1")
     await tv.send_command(SendRemoteKey.click("KEY_POWER"))
     async_connection.send.assert_called_once_with(
@@ -92,11 +89,11 @@ async def test_app_list(async_connection: Mock) -> None:
     """Ensure valid app_list data can be parsed."""
     async_connection.recv = Mock(
         side_effect=[
-            MS_CHANNEL_CONNECT_FUTURE,
-            ED_INSTALLED_APP_FUTURE,
+            create_future_with_result(MS_CHANNEL_CONNECT_SAMPLE),
+            create_future_with_result(ED_INSTALLED_APP_SAMPLE),
         ]
     )
-    async_connection.send = Mock(return_value=NONE_FUTURE)
+    async_connection.send = Mock(return_value=create_future_with_result(None))
     tv = SamsungTVWSAsyncRemote("127.0.0.1")
     await tv.start_listening()
     assert await tv.app_list() == [
@@ -122,12 +119,12 @@ async def test_app_list_bad_order(async_connection: Mock) -> None:
     """Ensure valid app_list data can be parsed, even if we get events in the wrong order."""
     async_connection.recv = Mock(
         side_effect=[
-            MS_CHANNEL_CONNECT_FUTURE,
-            ED_APPS_LAUNCH_FUTURE,
-            ED_INSTALLED_APP_FUTURE,
+            create_future_with_result(MS_CHANNEL_CONNECT_SAMPLE),
+            create_future_with_result(ED_APPS_LAUNCH_SAMPLE),
+            create_future_with_result(ED_INSTALLED_APP_SAMPLE),
         ]
     )
-    async_connection.send = Mock(return_value=NONE_FUTURE)
+    async_connection.send = Mock(return_value=create_future_with_result(None))
     tv = SamsungTVWSAsyncRemote("127.0.0.1")
     await tv.start_listening()
     assert await tv.app_list() == [
@@ -151,12 +148,15 @@ async def test_app_list_bad_order(async_connection: Mock) -> None:
 @pytest.mark.asyncio
 async def test_send_hold_key(async_connection: Mock) -> None:
     """Ensure simple data can be parsed."""
-    async_connection.recv = Mock(side_effect=[MS_CHANNEL_CONNECT_FUTURE])
-    async_connection.send = Mock(return_value=NONE_FUTURE)
+    async_connection.recv = Mock(
+        side_effect=[create_future_with_result(MS_CHANNEL_CONNECT_SAMPLE)]
+    )
+    async_connection.send = Mock(return_value=create_future_with_result(None))
 
     tv = SamsungTVWSAsyncRemote("127.0.0.1")
     with patch(
-        "samsungtvws.async_connection.asyncio.sleep", return_value=NONE_FUTURE
+        "samsungtvws.async_connection.asyncio.sleep",
+        return_value=create_future_with_result(None),
     ) as patch_sleep:
         await tv.send_command(SendRemoteKey.hold("KEY_POWER", 3))
 

